rust/rope/src/engine.rs

```diff
@@ -39,6 +39,9 @@ pub struct Engine {
 #[derive(Debug)]
 struct Revision {
     rev_id: usize,
+    /// The largest undo group number of any edit in the history up to this
```

Comment:
This works but my idea is to use the earliest revision number in an undo group as the (otherwise opaque) undo group identifier. Not 100% clear to me now which approach is better.